main/inc/lib/userportal.lib.php

```diff
@@ -1361,4 +1364,289 @@ public function return_hot_courses()
     {
         return CourseManager::return_hot_courses();
     }
+
+
+    /**
+     * UserPortal view for session, return the HTLK of the course list
+     * @param $user_id
+     * @return string
+     */
+    public function return_courses_and_sessions_view_by_session($user_id) {
+        $load_history = (isset($_GET['history']) && intval($_GET['history']) == 1) ? true : false;
+
+        if ($load_history) {
+            //Load sessions in category in *history*
+            $session_categories = UserManager::get_sessions_by_category($user_id, true);
+        } else {
+            //Load sessions in category
+            $session_categories = UserManager::get_sessions_by_category($user_id, false);
+        }
+
+        $html = '';
+
+        //Showing history title
+        if ($load_history) {
+            $html .= Display::page_subheader(get_lang('HistoryTrainingSession'));
+            if (empty($session_categories)) {
+                $html .=  get_lang('YouDoNotHaveAnySessionInItsHistory');
+            }
+        }
+
+        $special_courses = '';
+
+        $loadDirs = $this->load_directories_preview;
+
+        // If we're not in the history view...
+        $tabCoursesInfo = array();
+        if (!isset($_GET['history'])) {
+            // Display special courses
+            $special_courses = CourseManager::display_special_courses($user_id, $loadDirs);
+            // Display courses
+            $tabCourses = CourseManager::get_courses_list_by_user_id($user_id, false);  // [code=>xxx, real_id=>000]
+            foreach ($tabCourses as $i => $tabCourseCodeId) {
+                list($userCatId, $userCatTitle) = CourseManager::get_user_course_category_for_course($user_id, $tabCourseCodeId['real_id']);
+                $tabCourse = CourseManager::get_course_information_by_id($tabCourseCodeId['real_id']);
+                $tabCoursesInfo[] = array(
+                    'code' => $tabCourseCodeId['code'],
+                    'id' => $tabCourseCodeId['real_id'],
+                    'title' => $tabCourse['title'],
+                    'userCatId' => $userCatId,
+                    'userCatTitle' => $userCatTitle);
+            }
+            usort($tabCoursesInfo, 'self::cmpByCourse');
+        }
+
+        if (is_array($session_categories)) {
+            $tabCoursesInSession = SessionManager::get_named_session_course_for_coach($user_id); // all courses that are in a session
+        }
+
+        // we got all courses
+        // for each user category, sorted alphabetically, display courses
+        $tabUserCat = CourseManager::get_user_course_categories($user_id);
+        $tabCoursesAlreadyDisplayed = array();
+        uasort($tabUserCat, "self::cmpTabUserCat");
+        $tabUserCat[0] = '';
+
+        $html = '<div class="sessionViewBlock">';
+
+        foreach ($tabUserCat as $userCatId => $userCatTitle) {
+            // add user category
+            $htmlUC = '';
```

Comment:
Use more descriptive name variables example "$userCategoryHtml"  instead $htmlUC

$userCategories instead of $tabUserCat ..
$userCategoryTitle instead of $userCatTitle
etc..